observability-test/spanner.ts

```diff
@@ -815,8 +966,161 @@ describe('ObservabilityOptions injection and propagation', async () => {
         expectedEventNames,
         `Unexpected events:\n\tGot:  ${actualEventNames}\n\tWant: ${expectedEventNames}`
       );
+    });
+  });
+});
 
-      done();
+describe('Regression tests for fixed bugs', () => {
+  it('async/await correctly parents trace spans', async () => {
```

Comment:
Can we also add a test case using "callback"

Reply:
Yes please, and done, thank you @surbhigarg92, please take a look!